core/lib/src/config/builder.rs

```diff
@@ -21,7 +21,7 @@ pub struct ConfigBuilder {
     /// The secret key.
     pub secret_key: Option<String>,
     /// TLS configuration (path to certificates file, path to private key file).
-    pub tls: Option<(String, String)>,
+    pub tls: Option<(String, String, String)>,
```

Comment:
If the other TLS options can be specified without the mutual TLS option, then all three shouldn't be in a single `Option`. You should have another field just for mutual TLS that is an `Option`.